tests/test_api.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from typing import Optional
+
+import pytest
+from ghga_service_commons.api.testing import AsyncTestClient
+from hexkit.custom_types import JsonObject
+from hexkit.providers.mongodb.testutils import (  # noqa: F401
+    MongoDbFixture,
+    mongodb_fixture,
+)
+
+from mass.core import models
+from mass.main import get_configured_container, get_rest_api
+from tests.fixtures.config import get_config
+from tests.fixtures.joint import JointFixture, joint_fixture  # noqa: F401
+from tests.fixtures.mongo import populated_mongodb_fixture  # noqa: F401
+
+
+def compare(
+    *,
+    results: models.QueryResults,
+    count: int,
+    hit_length: int,
+    hits: Optional[list[models.Resource]] = None,
+    facets: Optional[list[models.Facet]] = None
+) -> None:
+    """Perform common comparisons for results"""
+    assert results.count == count
+    assert len(results.hits) == hit_length
+
+    if not facets:
+        configured_facets = (
+            get_config().searchable_classes["DatasetEmbedded"].facetable_properties
+        )
+        assert len(results.facets) == len(configured_facets)
+        assert {x.key for x in results.facets} == {x.key for x in configured_facets}
+    else:
+        assert results.facets == facets
+
+    if hits:
+        assert results.hits == hits
+
+
+async def call_search(
+    search_parameters: JsonObject, fixture: JointFixture
+) -> models.QueryResults:
+    """Convenience function to call the /rpc/search endpoint"""
+    response = await fixture.rest_client.post(url="/rpc/search", json=search_parameters)
+    results = models.QueryResults(**response.json())
+    return results
+
+
+@pytest.mark.asyncio
+async def test_search_options():
+    """Verify that we can request the configured resource class information correctly"""
+    config = get_config()
```

Comment:
better to get the config from the JointFixture

Reply:
Changes are in [5b9d8f1](https://github.com/ghga-de/mass/pull/7/commits/5b9d8f11d2a30d471df9b2de95c38af77d0d8f4a). The reason I had it like this though was because it didn't need the database or anything, so the long load time of the fixture felt unnecessary.

Reply:
OK, I see. That is also a valid choice.